### PR TITLE
v7.1.1-22 update

### PR DESCRIPTION
**Pull request by valgur**
This applies all changes made to the upstream build scripts up to version `7.1.1-22`. I also added all defines that were being set ing `config.h.in` but were missing in `config.h.cmake`.

I suggest you review these changes commit-by-commit. I tried to link to the relevant changes upstream for each.

I did not merge the actual upstream changes into this branch to keep things manageable. You can use my main branch at https://github.com/valgur/ImageMagick to try it out with everything already merged. I also included the updated conanfile.py from CCI for the dependencies there. As usual, you can use it by running `conan install .  --version 7.1.1-22 --build missing -o */*:shared=True` before building.

### CODE REVIEW

**Comment by Cyriuz** [comment]
Awesome, thank you for cleaning up a bit! I will take a look.

File: cmake/checks.cmake [hunk 1]
```diff
@@ -797,8 +753,8 @@ macro(magick_check_env)
   # Check `size_t' size
   check_type_size("size_t" SIZEOF_SIZE_T)
 
-  # Check `ssize_t' size
```
Comment (Cyriuz) [inline cmake/checks.cmake]: This should not be changed I think, and also the block below is needed for me to build on Windows.

File: CMakeLists.txt [hunk 2]
```diff
@@ -401,42 +426,42 @@ endif()
...
+
   # Read the version.h file so we can get some needed values using regex
   file(READ ${CMAKE_CURRENT_BINARY_DIR}/MagickCore/version.h CONTENT_VERSION_H)
-  
+
   # Try to find `MagickCopyright'
-  string(REGEX MATCH "\n#define[ ]+MagickCopyright[ ]+\"([^\"]+)\"\n" FOUND_MAGICK_COPYRIGHT ${CONTENT_VERSION_H})
+  string(REGEX MATCH "\n#define +MagickCopyright +\"([^\"]+)\"\n" FOUND_MAGICK_COPYRIGHT ${CONTENT_VERSION_H})
   if(FOUND_MAGICK_COPYRIGHT)
     set(MagickCopyright \"${CMAKE_MATCH_1}\")
```
Comment (Cyriuz) [inline CMakeLists.txt:451]: Remove quotes here too? All the quote removals break the ImageMagick.rc.cmake, but I can fix that if you want.

File: cmake/checks.cmake (hunk 1 above)
Comment (valgur) [reply cmake/checks.cmake]: Should be fixed now.

**Comment by valgur** [comment]
I further compared and fixed any discrepancies with the `config.h` generated by autotools on my local machine. They should match quite well now.

Next, I would like to:
- get tests working,
- add missing installation of config files under `/etc` and `/share`,
- maybe even run it in CI.

File: CMakeLists.txt (hunk 2 above)
Comment (valgur) [reply CMakeLists.txt:451]: Oops, I have not yet tested my changes on Windows. I definitely should.

File: cmake/checks.cmake [hunk 3]
```diff
@@ -101,11 +115,19 @@ macro(magick_check_env)
   check_include_file(dlfcn.h HAVE_DLFCN_H)
 
   # Check if `double_t' exists
```
Comment (Cyriuz) [inline cmake/checks.cmake:117]: I guess this comment should say float_t?

File: cmake/checks.cmake [hunk 4]
```diff
@@ -988,23 +949,6 @@ macro(magick_check_env)
     endif()
   endif()
 
-  # Check if <sys/types.h> doesn't define `ssize_t'
```
Comment (Cyriuz) [inline cmake/checks.cmake:991]: This block is still needed on Windows

**Pull request closed without merging**